### ISSUE TITLE
[DOC]: Better way to get WordPress intellisense

### ISSUE THREAD

**Issue by swashata** [bmewburn/intelephense-docs#2]
Hi,

First of all thank you very much for this awesome package. Your package is one of many reasons why I moved from Sublimetext to VSCode (don't mean to start a war here though).

I have been fiddling around a bit, and have been trying various methods to add WordPress intellisense. My preference has always been to just add the plugin/theme directory to vscode workspace. But in order to get intellisense of WordPress core functions, I had to add WordPress source in the workspace too. Not a great fan of this.

Then I came across [this](https://github.com/GiacoCorsiglia/wordpress-stubs) and since I use composer anyway, getting intellisense was a breeze.

I would've wanted to contribute to some doc, but since I didn't find any, I am opening an issue explaining the method and I leave it upto you to do what-so-ever you wish.

These are just the steps one need to follow to get WP intellisense.

1. First increase the default fileSize to 3MB or greater (VSCode > Preference > User Settings) `"intelephense.file.maxSize": 3000000`.
2. Now if you are using composer then `composer require --dev giacocorsiglia/wordpress-stubs`.
3. If you are not using composer then simply download [wordpress-stubs.php](https://github.com/GiacoCorsiglia/wordpress-stubs/blob/master/wordpress-stubs.php) and put it in your project directory (you probably don't want to track it with git).
4. Now from vscode command palette simply do a `Intelephense: Clear Cache and Reload`.
5. Also disable any WordPress core snippets extension you might have.

And from then on, it just works. Here's a video.

<a href="http://www.youtube.com/watch?feature=player_embedded&v=MEv9UlyaUPo
" target="_blank"><img src="http://img.youtube.com/vi/MEv9UlyaUPo/0.jpg" 
alt="WP Intelephense" width="560" height="315" border="10" /></a>

**Comment by xiaoyu-tamu** [comment]
set maxSize to 3200000 if giacocorsiglia/wordpress-stubs version is ^5.1

**Comment by szepeviktor** [comment]
Up-to-date WP stubs are available here: https://github.com/php-stubs/wordpress-stubs

**Comment by HowdyMcGee** [comment]
Is there a global way I can install this? I jump from WordPress project to project all day and installing this per project just doesn't seem feasible for me. Open to suggestions, otherwise I'll likely disable Intelephense entirely.

**Comment by vpjm** [comment]
I would also like to know if there is a global way to do this. Adding this to each project seems unnecessary. I don't like to have files that I don't need in my project directories. Is there any extension that can be installed to accomplish this?

**Comment by ntwb** [comment]
This was added in  1.3.3

See https://github.com/bmewburn/vscode-intelephense/issues/833#issuecomment-563984827

**Comment by nichols-green** [comment]
For those looking for an answer to this in 2021, simply install Intelephense, and in the settings, add `Core` and `Wordpress` to the `Intelephense: Stubs` section in preferences. 
![stubs](https://user-images.githubusercontent.com/82182892/129066336-472bf6f0-a395-410c-a691-3d9d30c1ecf9.PNG)
No need to add the stubs through composer or download.

**Comment by firecrauter** [comment]
Thanks nichols-green.
For those who don't know much about vscode, I'll save you the search: 
![image](https://user-images.githubusercontent.com/7044466/130083842-75115fda-6667-4374-bdf2-018c1aa600ab.png)
![image](https://user-images.githubusercontent.com/7044466/130084125-bc79239f-615f-475d-9599-bc5720a2de44.png)

**Comment by oliver-la** [comment]
I'd like to add my 2 cents here. I wanted to set up vscode for WordPress development, and one of the first extensions I installed was PHP Intelephense. It proved to be ridiculously difficult to find Intelephense already ships with WordPress support and it has to explicitly be enabled. Only learned about this when digging through multiple GitHub issues.

A great first step would be to link to bmewburn/intelephense-docs in the main repo of the vscode extension. I looked everywhere for a link to the documentation. 

Second, the documentation should include all optional features, with full descriptions. At least the premium features should be documented properly, there's no point in paying for features if you don't know how to use or enable them.

**Comment by DLX23** [comment]
possible to integrate acf (advanced custom fields) too?

**Comment by ntwb** [comment]
> possible to integrate acf (advanced custom fields) too?

Yes, ACF stubs seem available here, so install those with Composer and add it to the stubs list 

https://github.com/php-stubs?type=source

**Comment by oliver-la** [comment]
@ntwb I use this approach as well, and it works great! Wish there was a way to tell intelephense to search for globally installed stubs, so they don't have to be included in every project separately.

**Comment by szepeviktor** [comment]
> they don't have to be included in every project

I make many errors in my code (which I write ~once a month).
Using three default checks makes me more confident.
1. syntax errors
2. coding style
3. static analysis

I run my code after these do pass. (not before!)

**Comment by ItsMdYousuf** [comment]
> Thanks nichols-green. For those who don't know much about vscode, I'll save you the search: ![image](https://user-images.githubusercontent.com/7044466/130083842-75115fda-6667-4374-bdf2-018c1aa600ab.png) ![image](https://user-images.githubusercontent.com/7044466/130084125-bc79239f-615f-475d-9599-bc5720a2de44.png)

> Thanks a lot this is a very helpful trick.
